src/app/common/utils/resolvers/pages-wallet-details.resolver.ts

```diff
@@ -9,13 +9,21 @@ import { WalletsManagementItem } from 'src/app/pages/wallets/management/pages-wa
   providedIn: 'root',
 })
 export class PagesWalletDetailsResolver implements Resolve<WalletsManagementItem> {
-  public constructor(private readonly domainsWalletsGateway: DomainsWalletsGateway) {}
+  public constructor(private readonly domainsWalletsGateway: DomainsWalletsGateway, private readonly router: Router) {}
 
   public async resolve(route: ActivatedRouteSnapshot): Promise<WalletsManagementItem> {
     const wallets = await firstValueFrom(this.domainsWalletsGateway.getWallets());
     const id = route.paramMap.get('id')!;
     const wallet = wallets.find((wallet: IWalletApiResponse) => wallet.id === parseInt(id))!;
 
+    this.navigateErrorPageIfWalletNotFound(wallet);
```

Comment:
In that case you should stop executing rest of the code as wallet will be `undefined` so `return new WalletsManagementItem(wallet)` does not make sense